demos/whisper-base/whisper.js

```diff
@@ -209,19 +302,29 @@ export class Whisper {
         const { input_features } = await this.processor(audio_data);
         // -----------------------------------ENCODER INFERENCE-----------------------------------------
         // run encoder to get output
-        let encoder_inputs = {
+        const encoder_input = {
             type: this.dataType,
             data: input_features.data,
             dims: input_features.dims,
         };
         if (this.dataType == "float16") {
-            encoder_inputs.data = convertToFloat16OrUint16Array(encoder_inputs.data);
+            encoder_input.data = convertToFloat16OrUint16Array(encoder_input.data);
         }
         // log(`Pre-processing time: ${(performance.now() - start).toFixed(2)}ms`);
         // start = performance.now();
-        const { last_hidden_state } = await this.models["encoder"]["sess"].run({
-            input_features: new ort.Tensor(encoder_inputs.type, encoder_inputs.data, encoder_inputs.dims),
-        });
+        let encoder_outputs;
+        const encoder_inputs = {
+            input_features: new ort.Tensor(encoder_input.type, encoder_input.data, encoder_input.dims),
+        };
+        if (this.ioBinding) {
+            encoder_outputs = {
+                last_hidden_state: await this.createOutputMLTensor([1, 1500, 512], this.dataType),
```

Comment:
Is it reasonable/desireable to precreate the output tensor too? (if not, disregard)